### ISSUE TITLE
gradle sync too slow 

### ISSUE THREAD

**Issue by miguelbcr** [chrisjenx/Calligraphy#251]
Hi, I am experiencing that Gradle takes a long time synchronizing (about 10 min or more) when I add 

`compile 'uk.co.chrisjenx:calligraphy:2.1.0'`

but it does not happen with other libs. 
When I execute in terminal

`gradlew build --debug`

I get this:

`
18:24:58.996 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository] Detected non-existence of module 'pl.charmas.android:android-reactive-location:0.8' in resolver cache 'maven'
18:24:59.000 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository] Detected non-existence of module 'pl.charmas.android:android-reactive-location:0.8' in resolver cache 'maven2'
18:24:59.004 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository] Detected non-existence of module 'pl.charmas.android:android-reactive-location:0.8' in resolver cache 'maven3'
18:24:59.008 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.parser.IvyXmlModuleDescriptorParser] post 1.3 ivy file: using exact as default matcher
18:24:59.013 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository] Using cached module metadata for module 'pl.charmas.android:android-reactive-location:0.8' in 'MavenRepo'
18:24:59.017 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainDependencyResolver] Using pl.charmas.android:android-reactive-location:0.8 from Maven repository 'MavenRepo'
18:24:59.021 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Visiting dependency app_eatandcheck_android:presentation:unspecified(_debugCompile) -> uk.co.chrisjenx:calligraphy:2.1.0(dependency: uk.co.chrisjenx
#calligraphy;2.1.0 {compile=[default]})                                            
18:24:59.024 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.DependencyGraphBuilder] Selecting new module version uk.co.chrisjenx:calligraphy:2.1.0
18:24:59.027 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainDependencyResolver] Attempting to resolve uk.co.chrisjenx:calligraphy:2.1.0 using repositories [maven, maven2, maven3, MavenRepo, BintrayJCenter, maven4, maven5]
18:24:59.032 [DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading https://jitpack.io/uk/co/chrisjenx/calligraphy/2.1.0/calligraphy-2.1.0.pom
18:24:59.036 [DEBUG] [org.gradle.internal.resource.transfer.DefaultCacheAwareExternalResourceAccessor] Constructing external resource: https://jitpack.io/uk/co/chrisjenx/calligraphy/2.1.0/calligraphy-2.1.0.pom
18:24:59.040 [DEBUG] [org.gradle.cache.internal.btree.BTreePersistentIndexedCache] Opening cache artifact-at-url.bin (C:\Users\miguel\.gradle\caches\modules-2\metadata-2.15\artifact-at-url.bin)
18:24:59.057 [DEBUG] [org.gradle.internal.resource.transport.http.HttpResourceAccessor] Constructing external resource metadata: https://jitpack.io/uk/co/chrisjenx/calligraphy/2.1.0/calligraphy-2.1.0.pom
18:24:59.063 [DEBUG] [org.gradle.internal.resource.transport.http.HttpClientHelper] Performing HTTP HEAD: https://jitpack.io/uk/co/chrisjenx/calligraphy/2.1.0/calligraphy-2.1.0.pom
18:24:59.185 [DEBUG] [org.apache.http.impl.conn.PoolingClientConnectionManager] Connection request: [route: {s}->https://jitpack.io][total kept alive: 0; route allocated: 0 of 5; total allocated: 0 of 10]
18:24:59.198 [DEBUG] [org.apache.http.impl.conn.PoolingClientConnectionManager] Connection leased: [id: 0][route: {s}->https://jitpack.io][total kept alive: 0; route allocated: 1 of 5; total allocated: 1 of 10]
18:24:59.303 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnectionOperator] Connecting to jitpack.io:443
18:24:59.689 [DEBUG] [org.apache.http.client.protocol.RequestAddCookies] CookieSpec selected: best-match
18:24:59.703 [DEBUG] [org.apache.http.client.protocol.RequestAuthCache] Auth cache not set in the context
18:24:59.708 [DEBUG] [org.apache.http.client.protocol.RequestTargetAuthentication] Target auth state: UNCHALLENGED
18:24:59.713 [DEBUG] [org.apache.http.client.protocol.RequestProxyAuthentication] Proxy auth state: UNCHALLENGED
18:24:59.717 [DEBUG] [org.apache.http.impl.client.SystemDefaultHttpClient] Attempt 1 to execute request
18:24:59.721 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: HEAD /uk/co/chrisjenx/calligraphy/2.1.0/calligraphy-2.1.0.pom HTTP/1.1`
> Configuring > 4/4 projects > Resolving dependencies ':presentation:_debugCompile'

and the execution seems to be waiting for a long while. 
After that,  happens the same for:
> Configuring > 4/4 projects > Resolving dependencies ':presentation:_debugApk'

> Configuring > 4/4 projects > Resolving dependencies ':presentation:_releaseCompile'

I don't know what's happening.
Does anyone know how to solve it or how to get any clue about what happening? 

Thanks!

**Comment by chrisjenx** [comment]
@miguelbcr I'm not sure even where to start with this and suspicious that this is relating to Calligraphy at all.

Have you tried:
- Clearing your dependency cache.
- Using latest version of Gradle (2.11 at time of writing)
- Using the gradle wrapper
- Using Gradle daemon
- Checking you are using `jcenter()` `mavenCentral()` for your dependencies?

**Comment by miguelbcr** [comment]
Thanks a lot @chrisjenx !
I'm not an expert on gradle, but this is what I did before post the issue but without success:

- [ x] Clearing your dependency cache. **I removed `.gradle` folder on my user folder system**
- [ ] Using latest version of Gradle (2.11 at time of writing)  
- [ ] Using the gradle wrapper. **I don't know exactly what is it and how to use it. I used `gradlew `on the terminal...**
- [ ] Using Gradle daemon. **I don't know exactly what is it and how to use it.**
- [ x] Checking you are using jcenter() mavenCentral() for your dependencies?  **Yes, I have both them in  `build.gradle` files**


So finally, It works fine updating gradle version from 2.4 to 2.11 (File > project structure > project > gradle version). It took a while to download that version (saved within `.gradle\wrapper\dists` folder on my user folder system). After I cleaned and built the project, and all went fast, as usual.

Thanks!
